### ISSUE TITLE
check_subrc, skip check if assigning and checking assigning, dynamic assignment

### ISSUE THREAD

**Issue by victorizbitskiy** [abaplint/abaplint#1863]
should not report error for:

```abap
DATA: lcl_ref TYPE REF TO data.
FIELD-SYMBOLS: <restab_standatd> TYPE STANDARD TABLE,

ASSIGN: lcl_ref->* TO <restab_standatd>.
IF <restab_standatd> IS NOT ASSIGNED.
ENDIF.
```

**Comment by larshp** [comment]
@victorizbitskiy try again, 2.71.5 published

**Comment by victorizbitskiy** [comment]
@larshp for the case above, everything is fine, but there is one more case:
```abap 
FIELD-SYMBOLS <fs> TYPE any.
DATA lv_char5 TYPE c LENGTH 5.
DATA lv_offset TYPE i.

lv_char5 = 'test1'.
lv_offset = 5.

ASSIGN lv_char5+lv_offset(1) TO <fs>.
IF sy-subrc = 0.
  RETURN.
ENDIF.
```
Is it possible to make a warning appear? In this case, we need to check not the sy-subrc, but the IS ASSIGNED.

**Comment by larshp** [comment]
I cannot reproduce the issue, the rule https://rules.abaplint.org/check_subrc/ checks for subrc
![image](https://user-images.githubusercontent.com/5888506/111269673-3377b480-862f-11eb-8ebb-3cab06bcfe9e.png)

Are you suggesting to have a check where IS ASSIGNED is preferred instead of checking SY-SUBRC?

**Comment by victorizbitskiy** [comment]
> Are you suggesting to have a check where IS ASSIGNED is preferred instead of checking SY-SUBRC?

Yes, right. For example, as in SLIN.

![image](https://user-images.githubusercontent.com/62376906/111271466-4647b480-8642-11eb-82c3-10b13d30773e.png)

**Comment by larshp** [comment]
Oh, I thought ASSIGN always set SY-SUBRC, heh, always another twist in ABAP

https://help.sap.com/doc/abapdocu_750_index_htm/7.50/en-US/abapassign.htm

**Comment by victorizbitskiy** [comment]
Yes, until recently I thought so too 😅

**Comment by larshp** [comment]
something like this, these are okay,

```abap
  ASSIGN foo TO <left_operand>.
  IF <left_operand> IS ASSIGNED.
  ENDIF.

  ASSIGN (name) TO <left_operand>.
  IF sy-subrc = 0.
  ENDIF.

  ASSIGN dref->* TO <left_operand>.
  IF sy-subrc = 0.
  ENDIF.

  ASSIGN dobj INCREMENT inc TO <left_operand>.
  IF sy-subrc = 0.
  ENDIF.

  ASSIGN COMPONENT comp OF STRUCTURE struc TO <left_operand>.
  IF sy-subrc = 0.
  ENDIF.
```

but these are errors?

```abap
  ASSIGN foo TO <left_operand>.
  IF sy-subrc = 0.
  ENDIF.

  ASSIGN lcl_ref->* TO <restab_standatd>.
  IF <restab_standatd> IS NOT ASSIGNED.
  ENDIF.
```

**Comment by victorizbitskiy** [comment]
That's a complicated question... Here's what I think: 

1. The first case refers to the static variant. 
> ...The value of the system field in this case is always the same as it was before the ASSIGN statement was executed and therefore does not indicate whether the statement was successful.

https://help.sap.com/doc/abapdocu_750_index_htm/7.50/en-US/abapassign_mem_area_static_dobj.htm
So I think this sy-subrc check is an error.

2. The second case refers to the dynamic variant. In these variants, the statement ASSIGN sets the return code sy-subrc, but I think the ASSIGNED / NOT ASSIGNED check is also valid if the field symbol was UNASSIGN'ed before the assignment. Like this:

```abap
UNASSIGN <restab_standatd>.

ASSIGN: lcl_ref->* TO <restab_standatd>.
IF <restab_standatd> IS NOT ASSIGNED.
ENDIF.
```

Sorry for my English. I am still using the translator...😅
